app/messaging/application/process-application.js

```diff
@@ -19,19 +34,21 @@ const processApplication = async (msg) => {
     console.log(`Application received : ${JSON.stringify(applicationData)} with sessionID ${sessionId} and messageID ${messageId}.`)
 
     const existingApplication = await applicationRepository.getBySbi(
-      applicationData.organisation.sbi // todo consider reworking this for re application logic as this could pull back more than one agreement
+      applicationData.organisation.sbi
     )
 
     if (
       existingApplication &&
-      existingApplication.statusId !== applicationStatus.withdrawn &&
-      existingApplication.statusId !== applicationStatus.notAgreed
-      // todo consider being able to apply again within 10 months
+      ((existingApplication.statusId !== applicationStatus.withdrawn &&
+      existingApplication.statusId !== applicationStatus.notAgreed &&
+      // check if it passes 10 month rule here and chuck error if it doesn't
+      isPastTimeLimit(timeLimitDates(existingApplication)) === false) ||
+      existingApplication.statusId === applicationStatus.agreed)
```

Comment:
Does this say that if the last application is agreed that they cannot apply again nNo matter the timeframe?

Reply:
Yes. thats right